docs/source/blueprints/BP005.rst

```diff
@@ -0,0 +1,291 @@
+   will make sure they will be well suited for different use cases.  By having
+   more users, the quality and collaboration of those libraries will be improved.
+   Also, third-party software will be able to import those libraries without the
+   need of having the entire Avocado framework as a dependency. This will reduce
+   the learning curve of future potential users that don't need the entire
+   framework.
+
+3. Making it clear to our users that Avocado framework is not Python-centric:
+
+   Currently, those utility libraries can only be used by Python tests and the
+   Avocado framework was designed to support tests written in multiple
+   languages.  By having those libraries outside the core harness codebase we
+   reinforce the language agnostic characteristic of Avocado.
+
+4. Reduce traffic in the main framework repository and allow for a scalable
+   community on its own standalone project
+
+   As evidenced by “avocado-misc-tests”, a more focused standalone project will
+   attract a larger number of similar minded developers and users. By being a
+   simpler standalone repository, with its own specific contributions
+   guidelines, the entry barriers are also lowered.
+
+
+Specification
+*************
+
+To avoid the need to change our current imports, our first attempt was to keep
+the "avocado.utils" namespace regardless of the location of its codebase by
+using the Implicit Package Namespace (PEP-420) approach. Namespace packages are
+a mechanism for splitting a single Python package across multiple directories
+on disk. Unfortunately, to be able to convert the avocado.utils.* modules and
+packages into a namespace package, we would need to remove the `__init__.py`
+files from multiple places. That would break, among other things, the current
+Avocado egg based deployment model used on spawners such as podman.
+
+Because of that, we decided to have a new namespace for the utility libraries.
+We ran a poll to choose the name and `autils` (Avocado Utils) was the winner.
+So our goal is to host common libraries under `avocado-project/autils`
+repository on GitHub and make them also available on PyPI under `python-autils`
+package name, since the `autils` name is unavailable.
+
+Utility onboarding
+==================
+
+The primary goal of autils is to be a central place for modules that ease the
+interaction with system level features and interfaces.  As such, if a user
+needs to interact with such a system level feature or interface, he/she may
+propose a new module.
+
+For instance, if a user finds himself/herself constantly interacting with disk
+partitions, he/she may want to propose a “partitions” module that provides
+basic APIs for that. In this case it makes sense to have it as part of this
+central repository.
+
+However, if the utility intends to ease the interaction with an esoteric
+feature, system or device, this may not be a good place for it.
```

Comment:
> I kind of understand the cases this comment against, but I just wonder if we can explain more on that. My point is we need to avoid that it would be acted by fully depending on maintainers' preference.

I believe the original meaning of this text is one in which a user writes a module that can interact with, for instance, a very specific hardware device that very few people will have access to.

For instance, let's suppose autils receives a PR like this:

---

*Subject:* Introduce autils.usrobotics

*Description:*

_This PR introduces support for manipulating switches for the Dial-Up Modems made by US Robotics in the 1990s.  Supported models include: XXXX, and YYYY._

---

The audience of the module would be extremely limited, and it would be close to impossible to have it tested on our CI.

I know the example is a bit extreme, but the same example could apply to out-of-tree, non-mainline kernel features.

But yeah, I think we can expand that topic with some examples.

Reply:
Thanks @luckyh , I have added a new sentence at the end of that paragraph, so we can clarify this could be queried before submission in a GH thread. I will send a v2.

I'm resolving this conversation so we can approve this Blueprint.